raiden-dapp/src/components/AmountInput.vue

```diff
@@ -63,79 +42,82 @@ export default class AmountInput extends Vue {
   token?: Token;
   @Prop({ default: false, type: Boolean })
   limit!: boolean;
-  @Prop({ default: false, type: Boolean })
-  light!: boolean;
-  @Prop({ default: false, type: Boolean })
-  padded!: boolean;
+  @Prop({ default: '0.0', type: String })
+  placeholder!: string;
+  @Prop({ required: false, default: () => Zero })
+  max!: BigNumber;
 
   valid: boolean = true;
-  amount: string = '0.00';
-  private static numericRegex = /^\d*[.,]?\d*$/;
+  amount: string = '';
+  private static numericRegex = /^\d*[.]?\d*$/;
 
   readonly rules = [
     (v: string) => {
       return !!v || this.$parent.$t('amount-input.error.empty');
     },
     (v: string) =>
       !this.limit ||
-      this.noDecimalOverflow(v) ||
+      (v && this.noDecimalOverflow(v)) ||
       this.$parent.$t('amount-input.error.too-many-decimals', {
         decimals: this.token!!.decimals
       }),
     (v: string) =>
       !this.limit ||
-      this.hasEnoughBalance(v) ||
+      (v && this.hasEnoughBalance(v, this.max)) ||
       this.$parent.$t('amount-input.error.not-enough-funds', {
-        funds: this.token!!.units
+        funds: BalanceUtils.toUnits(this.max, this.token!!.decimals),
+        symbol: this.token!!.symbol
       })
   ];
 
   private noDecimalOverflow(v: string) {
     return (
-      v &&
       AmountInput.numericRegex.test(v) &&
-      !BalanceUtils.decimalsOverflow(v, this.token!!)
+      !BalanceUtils.decimalsOverflow(v, this.token!!.decimals)
```

Comment:
It these `!!` the non-null assertion operator? If so, I think a single `!` is enough

Reply:
Comment fixed, thank you.